packages/react-reconciler/src/ReactFiber.js

```diff
@@ -345,7 +346,15 @@ export function createWorkInProgress(
 }
 
 export function createHostRootFiber(isAsync: boolean): Fiber {
-  const mode = isAsync ? AsyncMode | StrictMode : NoContext;
+  let mode = isAsync ? AsyncMode | StrictMode : NoContext;
+
+  if (enableProfilerTimer && isDevToolsPresent) {
+    // Always collect profile timings when DevTools are present.
```

Comment:
This only affects the profiling bundle, right?

At first I was a bit worried we'd do this in production mode too, but now I realized that we seemingly don't. Then it should be fine.

In general we try to avoid any extra overhead based on DevTools *presence* (i.e. until we actually connect). But a bit of overhead in profiling bundle specifically (and not in production) seems okay to me.

Reply:
Yes, it only affects profiling builds.